storage/blockdevice/tests/UNITTESTS/blockdevice/HeapBlockDevice/CMakeLists.txt

```diff
@@ -0,0 +1,67 @@
+# Copyright (c) 2021 ARM Limited. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+set(TEST_NAME heap-blockdevice-unittest)
+
+add_executable(${TEST_NAME})
+
+target_compile_definitions(${TEST_NAME} 
+    PRIVATE
+        MBED_CONF_FAT_CHAN_FFS_DBG=0
+        MBED_CONF_FAT_CHAN_FF_FS_READONLY=0
+        MBED_CONF_FAT_CHAN_FF_FS_MINIMIZE=0
+        MBED_CONF_FAT_CHAN_FF_USE_STRFUNC=0
+        MBED_CONF_FAT_CHAN_FF_USE_FIND=0
+        MBED_CONF_FAT_CHAN_FF_USE_MKFS=1
+        MBED_CONF_FAT_CHAN_FF_USE_FASTSEEK=0
+        MBED_CONF_FAT_CHAN_FF_USE_EXPAND=0
+        MBED_CONF_FAT_CHAN_FF_USE_CHMOD=0
+        MBED_CONF_FAT_CHAN_FF_USE_LABEL=0
+        MBED_CONF_FAT_CHAN_FF_USE_FORWARD=0
+        MBED_CONF_FAT_CHAN_FF_CODE_PAGE=437
+        MBED_CONF_FAT_CHAN_FF_USE_LFN=3
+        MBED_CONF_FAT_CHAN_FF_MAX_LFN=255
+        MBED_CONF_FAT_CHAN_FF_LFN_UNICODE=0
+        MBED_CONF_FAT_CHAN_FF_LFN_BUF=255
+        MBED_CONF_FAT_CHAN_FF_SFN_BUF=12
+        MBED_CONF_FAT_CHAN_FF_STRF_ENCODE=3
+        MBED_CONF_FAT_CHAN_FF_FS_RPATH=1
+        MBED_CONF_FAT_CHAN_FF_VOLUMES=4
+        MBED_CONF_FAT_CHAN_FF_STR_VOLUME_ID=0
+        MBED_CONF_FAT_CHAN_FF_VOLUME_STRS=\"RAM\",\"NAND\",\"CF\",\"SD\",\"SD2\",\"USB\",\"USB2\",\"USB3\"
+        MBED_CONF_FAT_CHAN_FF_MULTI_PARTITION=0
+        MBED_CONF_FAT_CHAN_FF_MIN_SS=512
+        MBED_CONF_FAT_CHAN_FF_MAX_SS=4096
+        MBED_CONF_FAT_CHAN_FF_USE_TRIM=1
+        MBED_CONF_FAT_CHAN_FF_FS_NOFSINFO=0
+        MBED_CONF_FAT_CHAN_FF_FS_TINY=1
+        MBED_CONF_FAT_CHAN_FF_FS_EXFAT=0
+        MBED_CONF_FAT_CHAN_FF_FS_HEAPBUF=1
+        MBED_CONF_FAT_CHAN_FF_FS_NORTC=0
+        MBED_CONF_FAT_CHAN_FF_NORTC_MON=1
+        MBED_CONF_FAT_CHAN_FF_NORTC_MDAY=1
+        MBED_CONF_FAT_CHAN_FF_NORTC_YEAR=2017
+        MBED_CONF_FAT_CHAN_FF_FS_LOCK=0
+        MBED_CONF_FAT_CHAN_FF_FS_REENTRANT=0
+        MBED_CONF_FAT_CHAN_FF_FS_TIMEOUT=1000
+        MBED_CONF_FAT_CHAN_FF_SYNC_t=HANDLE
+        MBED_CONF_FAT_CHAN_FLUSH_ON_NEW_CLUSTER=0
+        MBED_CONF_FAT_CHAN_FLUSH_ON_NEW_SECTOR=1
+)
+
+target_sources(${TEST_NAME}
+    PRIVATE
+        ${MBED_PATH}/storage/blockdevice/source/HeapBlockDevice.cpp
```

Comment:
If we link against a library that provides a heap block device instead of listing the sources here, we'll help move Mbed to be more modular. The config you've added in `target_compile_definitions()` in this file would move to the library target and wouldn't need to be duplicated.

Reply:
Ideally, we only have to include the source under test only that test covers but unfortunately, some unit test includes all the source c/cpp files instead of using storage stubs which is present in mbed-os/UNITTESTS/stubs, the improvement could rework on this test to remove cpp files which are irrelevant to the test